src/Common/Core/Impl/OS/Win32NativeEnvironmentBlock.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using static Microsoft.Common.Core.NativeMethods;
+
+namespace Microsoft.Common.Core.OS {
+    public class Win32NativeEnvironmentBlock : IDisposable {
+        public IntPtr NativeEnvironmentBlock { get; }
+
+        public Win32NativeEnvironmentBlock(IntPtr env) {
+            NativeEnvironmentBlock = env;
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing) {
```

Comment:
I used the Implement Disposable pattern that C# generates. What is the change you are suggesting? the pointer that is used here is obtained from AllocHGlobal when i convert the environment key-value pair back to native.